### PR TITLE
Introducing InstrumentDesign and OpticalDesign

### PR DESCRIPTION
**Pull request by jacobdparker**
Introducing two new classes `InstrumentDesign` and `OpticalDesign` which make it easier to store configuration defaults and different MOXSI design iterations.

### CODE REVIEW

File: mocksipipeline/detector/design.py [hunk 1]
```diff
@@ -49,20 +53,97 @@ class InstrumentDesign:
...
+    grating_roll_angle=0 * u.deg,
 )
+
+
+@dataclasses.dataclass
+class InstrumentDesign:
+    channel_list: list[Channel]
+
+
+# These are based off the current instrument design
```
Comment (wtbarnes) [inline mocksipipeline/detector/design.py:87]: It'd probably be better to put all of the actual instances into a separate file.

**Comment by wtbarnes** [comment]
Looks like the tests are failing because the channel classes are instantiated differently now. Should just be a matter of fixing the tests to instantiate them correctly with an `OpticalDesign` and a filter choice.

**Comment by wtbarnes** [comment]
Alright I've reorganized things by renaming `detector` to `instrument` and moving the optical design stuff to `optics`. Anything that gets instantiated gets put in a `configuration.py` file. 

Now that I'm seeing the length of one of these instrument configurations, I wonder if it makes sense to actually make `instrument.configuration` a subpackage and then put each configuration in its own file. The optics configurations could stay as they are since they're relatively short.

**Comment by jacobdparker** [comment]
Yeah, I can see that.  It is very clunky that they are so long, but I do think having the math there is useful as a record.  Otherwise they just seem like magic numbers.

**Pull request merged**